src/caffe/layers/base_data_layer.cpp

```diff
@@ -77,7 +77,7 @@ void BasePrefetchingDataLayer<Dtype>::InternalThreadEntry() {
 #ifndef CPU_ONLY
   cudaStream_t stream;
   if (Caffe::mode() == Caffe::GPU) {
-    CUDA_CHECK(cudaStreamCreateWithFlags(&stream, cudaStreamNonBlocking));
+    CUDA_CHECK(cudaStreamCreate(&stream));
```

Comment:
Simon wanted to move away from non-blocking streams. I don't think there is any synchronization issue, it was working both with and without blocking streams.